src/workerd/api/sockets.c++

```diff
@@ -9,8 +9,41 @@
 namespace workerd::api {
 
 
+bool isValidAddress(kj::StringPtr address) {
```

Comment:
Should likely add a comment here that we're only determining validity in terms of length and characters. There are other factors that may make an address invalid. For example `foo:bar::baz_` would pass this algorithm but is not a valid domain.